### ISSUE TITLE
Why not wait for cache sync before run the controller worker?

### ISSUE THREAD

**Issue by huskysun** [GoogleCloudPlatform/spark-on-k8s-operator#879]
[In this doc](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/controllers.md) (it's the No.6 one), it recommends that a k8s controller should wait for cache sync before the worker starts. However, in the implementation it seems that the code is not following this guideline (https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/pkg/controller/sparkapplication/controller.go#L163) - it starts all workers before the cache is synced. Is there any particular reason for that? I'm not very familiar with this, just curious about the discrepancy between the k8s doc and spark operator implementation. Thanks!

**Comment by liyinan926** [comment]
It was the case before. But somehow the order was switched in https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/291/files#diff-b7eea120d7ac0e8587781b863b7db8e4. @akhurana001 do you remember why it was switched?

**Comment by akhurana001** [comment]
I don't remember a particular reason and the PR was a while ago. However, I think it should be safe to swap them. I can do that shortly in another PR.

**Comment by huskysun** [comment]
Yeah I think we should better follow the guideline and wait the cache sync before starting the workers if there is no particular reason to do otherwise.

**Comment by liyinan926** [comment]
Fixed in #880.
